torchdata/dataloader2/reading_service.py

```diff
@@ -527,3 +544,27 @@ def restore(self, datapipe, serialized_state: bytes) -> DataPipe:
             else:
                 warnings.warn(f"{rs} doesn't support `restore` from state, skipping...")
         return datapipe
+
+    def _pause(
+        self, pause_fn: Optional[Callable[[DataPipe], DataPipe]] = None
+    ) -> Optional[Callable[[DataPipe], DataPipe]]:
+        for rs in self.reading_services:
+            if hasattr(rs, "_pause"):
+                pause_fn = rs._pause(pause_fn)
+        return pause_fn
```

Comment:
I think we need an explanation of why this needs to accept an initial `pause_fn`? I assume it must be passed in from `DataLoader2._pause()`? And then the return object `pause_fn` will be used for the subsequent `rs._pause()`?

Ditto for `_resume` and `_limit`.

Reply:
And, the general reason that there is an initial pause_fn is to have an aligned API across `_pause`.